hass_nabucasa/files.py

```diff
@@ -111,6 +112,20 @@ async def upload(
             )
 
             self._do_log_response(response)
+            if response.status == 400:
+                # We can try to get some context.
+                error = await response.text()
+                if error and "<Message>" in error and "</Message>" in error:
+                    with contextlib.suppress(AttributeError, IndexError):
+                        # This is ugly but it's the best we can do, we have no control
+                        # over the error message structure, so we try what we can.
+                        parsed = (
+                            error.split("<Message>")[1]
+                            .split("</Message>")[0]
+                            .replace("\n", " ")
+                        )
+                        raise FilesError(f"Failed to upload: (400) {parsed}")
```

Comment:
If we fail parsing, don't we want to include the response in the raised `FileError` to aid debugging?

Reply:
We do, that is handled when we raise for status

Reply:
The stats is raised in the exception, and the entire response text is dumped in the log.